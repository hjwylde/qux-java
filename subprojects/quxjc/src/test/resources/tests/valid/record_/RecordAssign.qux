package valid.record_

{int x, str y} f({int x, str y} xs):
    xs.x = 3

    return xs

void main([str] args):
    xs = {x: 5, y: 'foo'}
    print xs.x
    print xs.y
    print f(xs).x
    print xs.x
    xs.y = 'bar'
    print xs
    xs.y = null
    print xs
    xs.x *= 3
    print xs
    xs.x += 0
    print xs
    xs.x = {y: {z: null}}
    print xs
    xs.x.y.a = null
    print xs

