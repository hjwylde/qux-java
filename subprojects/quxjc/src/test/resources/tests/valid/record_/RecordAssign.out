5
foo
3
5
{x: 5, y: bar}
{x: 5, y: null}
{x: 15, y: null}
{x: 15, y: null}
{x: {y: {a: null, z: null}}, y: null}
{x: 5, y: 6}
{x: foo, y: 6}
